plugin/trino-hudi/src/main/java/io/trino/plugin/hudi/model/HudiReplaceCommitMetadata.java

```diff
@@ -27,14 +28,15 @@
 @JsonIgnoreProperties(ignoreUnknown = true)
 public class HudiReplaceCommitMetadata
 {
-    private Map<String, List<String>> partitionToReplaceFileIds;
-    private Boolean compacted;
+    private final Map<String, List<String>> partitionToReplaceFileIds;
+    private final Boolean compacted;
 
-    // for ser/deser
-    public HudiReplaceCommitMetadata()
+    @JsonCreator(mode = JsonCreator.Mode.PROPERTIES)
+    public HudiReplaceCommitMetadata(@JsonProperty("partitionToReplaceFileIds") final Map<String, List<String>> partitionToReplaceFileIds,
+                                     @JsonProperty("compacted") final Boolean compacted)
     {
-        partitionToReplaceFileIds = ImmutableMap.of();
-        compacted = false;
+        this.partitionToReplaceFileIds = partitionToReplaceFileIds;
+        this.compacted = compacted;
```

Comment:
Are these fields nulalble?

Reply:
They are not. If eg. `partitionToReplaceFileIds` become null this code will throw an NPE (from `HudiActiveTimeline::reetFileGroupsReplaced`): 
```scala
                        // get replace instant mapping for each partition, fileId
                        return replaceMetadata.getPartitionToReplaceFileIds().entrySet().stream()
                                .flatMap(entry -> entry.getValue().stream().map(fileId ->
                                        Map.entry(new HudiFileGroupId(entry.getKey(), fileId), instant)));
```

Reply:
Then, it would be better to add `ImmutableMap.copyOf` & `requireNonNull` and change `Boolean` to `boolean`
```
        this.partitionToReplaceFileIds = ImmutableMap.copyOf(requireNonNull(partitionToReplaceFileIds, "partitionToReplaceFileIds is null"));
```